src/towncrier/build.py

```diff
@@ -114,51 +114,47 @@ def __main(
     to_err = draft
 
     click.echo("Loading template...", err=to_err)
-    with open(config["template"], "rb") as tmpl:
+    with open(config.template, "rb") as tmpl:
         template = tmpl.read().decode("utf8")
 
     click.echo("Finding news fragments...", err=to_err)
 
-    definitions = config["types"]
-
-    if config.get("directory"):
-        fragment_base_directory = os.path.abspath(config["directory"])
+    if config.directory:
```

Comment:
I wonder if the behavior of this conditional is intentional or if the following would be better:

```suggestion
    if config.directory is not None:
```

Reply:
I wondered the same thing, but I tried to stay away from functional changes as well as I could. If our types are correct, it should indeed work your way!